src/controller/trainee/Controller.ts

```diff
@@ -1,138 +1,153 @@
 import { Request, Response, NextFunction } from 'express';
 import * as jwt from 'jsonwebtoken';
 import config from '../../config/configuration';
+import TraineeRepository from '../../repositories/trainee/TraineeRepository';
 
-class Trainee {
-  getTrainee(req: Request, res: Response, next: NextFunction) {
+const traineeRepository: TraineeRepository = new TraineeRepository();
+
+class TraineeController {
+  /**
+   * @description: Get Trainee data based on Token
+   * @param req
+   * @param res
+   * @param next
+   * @returns One trainee data
+   */
+  async get(req: Request, res: Response, next: NextFunction) {
     try {
-      const trainee = [
-        {
-          _id: 1,
-          name: 'Mukesh',
-          address: 'Noida',
-          role: 'SD1',
-        },
-        {
-          _id: 2,
-          name: 'Rajeev',
-          address: 'New Delhi',
-          role: 'HR',
-        },
-        {
-          _id: 3,
-          name: 'Narayan Murti',
-          address: 'Bengaluru',
-          role: 'HOD',
-        },
-      ];
-      return res
-        .status(200)
-        .send({ message: 'User Fetched Successfully', data: trainee });
+      const {skip = 0, limit = 10} = req.query
+      const role = {role:"trainee"};
+      const token = req.header('Authorization');
+      if (!token) {
+        return next({ err: 'Unauthorized', message: 'Token not found', status: 403 });
+      }
+      const { secret } = config;
+      let trainee = jwt.verify(token, secret);
+      if (!trainee) {
+        return next({ err: 'Unauthorized', message: 'You are not allowed', status: 403 });
+      }
+      const traineeCount = await traineeRepository.countData()
+      const traineeData = await traineeRepository.findData({ skip, limit, role });
+      const data = [{traineeCount, traineeData}]
+      return res.status(200).send({
+        message: 'trainee data fetched successfully',
+        data: data,
+      });
     } catch (error) {
-      return res.status(500).send({ err: error, message: 'error' });
+      return res.status(500).send({ err: error, message: 'Something went wrong..!!' });
     }
   }
-  // adding new trainee data (POST)
-  addTrainee(req: Request, res: Response, next: NextFunction) {
+  
+  
+  /**
+   * @description CREATE TRAINEE BY ID
+   * @param req 
+   * @param res 
+   * @param next 
+   * @returns New Trainee data
+   */
+  async create(req: Request, res: Response, next: NextFunction) {
     try {
-      const data = [];
       const newTrainee = {
-        _id: req.body._id ? req.body._id : Math.floor(Math.random() * 100),
+        id: req.body.id,
         name: req.body.name,
-        email:req.body.email,
-        address: req.body.address,
-        role: req.body.role,
+        email: req.body.email,
+        password: req.body.password,  
+        role: req.body.role ? req.body.role : 'trainee',
       };
```

Comment:
id is not entered during creation, it is automatically generated. if not then fix your method working